docs/guides/dependency-installation.rst

```diff
@@ -1,229 +1,245 @@
 Dependency Installation
 =======================
 
-Many tasks will require external sources and dependencies in order to function. These are specified on a per-task basis by passing import arguments to the ``@sdk.task()`` decorator. Multiple dependency import arguments can be used simultaneously.
-
-For convenience the table below summarises which importer should be used in various contexts. The following sections give a more complete explanation of these importers and their usage.
-
-.. list-table:: Quick Reference
-    :widths: 25 25 25 25
-    :header-rows: 1
-
-    * - Importing From
-      - Importer
-      - Runtime
-      - Notes
-    * - GitHub repo
-      - ``GithubImport``
-      - remote
-      - On Compute Engine, secrets can be used to access private repos.
-    * - Non-GitHub git repo
-      - ``GitImport``
-      - remote
-      -
-    * - PyPi
-      - ``PythonImports``
-      - remote, local
-      -
-    * - Jupyter Notebook
-      - ``InlineImport``
-      - remote, local
-      - Actually having to specify an ``InlineImport`` is rare as this is the default import when running in a Jupyter Notebook. There are occasional cases where this should be used outside of Notebooks, see the section below for details.
-    * - Local - file within this repo.
-      - None
-      - local
-      - This is the default in all non-interactive sessions, so does not need to specified explicitly.
+If you send your workflow for execution on a remote runtime (Orquestra Platform), you have to somehow move your task code to the server.
+It's not always a straightforward goal and there are a couple of techniques to achieve it, each one comes with trade-offs.
 
-Good Practice
--------------
+In Orquestra, you declare how to supply the task's code and the libraries it depends on using *Imports*.
+These are specified on a per-task basis by setting ``source_import=`` and ``dependency_imports=`` parameters in the ``@sdk.task()`` decorator.
 
-For workflows running remotely the ``GithubImport`` importer should be used unless there is a specific reason to do otherwise (i.e. the code is hosted somewhere other than GitHub). ``GitImport.infer()`` is highly user-specific and so should never be used in a collaborative context.
 
-For workflows running locally, the task decorator can typically be used without specifying the source importer as the SDK will set this appropriately for the context:
+Quick Reference
+===============
 
-.. code-block:: python
+For convenience the table below summarizes which *Import* should be used in various contexts.
+The following sections give a more complete explanation of these importers and their usage.
 
-    @sdk.task
-    def my_task():
-        pass
 
-will perform a local import unless running in a Jupyter Notebook (or other interactive session) in which case an inline import will be performed.
+.. list-table:: Quick Reference For Orquestra Imports
+   :header-rows: 1
 
+   * - Import
+     - Pros
+     - Cons
```

Comment:
Suggestion: I feel like "pros" and "cons" lacks some intentionality on our part. How would you feel about reframing this as something like "for use with" and "be aware"?